fusionjs/fusion-cli
# Upgrade to webpack@4

This upgrades webpack to version 4 in a way that keeps things mostly the same as what's in master for the easiest possible upgrade. While it may be possible to update our usage of webpack plugins now that webpack 4 has new interfaces, decoupling the two efforts will allow us to move more quickly than doing everything at once.

**Changes:**

* Update plugins to use new hook API interface.
* Update tests to remove '-e' (causes webpack to infinitely re-bundle).
* Update instrumented import plugin to use moduleNamespacePromise and chunkGroup interface.
* Greenkeep dependencies.

**Test plan:**

* Ensure that existing tests pass.
* Verify fusion-cli works in the browser-test repo.
* Verify fusion-cli works in various boilerplate repos.
* Inspect filesystem for expected dynamic import instrumented bundles.

Fixes #264
